### PR TITLE
Add virtual columns `_block_offset` and `_data_version` for `MergeTree` tables

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added persistent virtual column `_block_offset` which stores original number of row in block that was assigned at insert. Persistence of column `_block_offset` can be enabled by setting `enable_block_offset_column`. Added virtual column`_part_data_version` which contains either min block number or mutation version of part.  Persistent virtual column `_block_number` is not considered experimental anymore.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

Depends on #60205.

### CODE REVIEW

**Review by alesapin — APPROVED** [review]

**Comment by CurtizJ** [comment]
Test failures are related to new analyzer.

**Comment by rschu1ze** [comment]
Hmm, is it intentional that `enable_block_offset_column` and `enable_block_number_column` are not mentioned in the user docs?

File: src/Storages/MergeTree/MergeTreeSettings.h [hunk 1]
```diff
@@ -186,6 +186,8 @@ struct Settings;
     M(Bool, disable_freeze_partition_for_zero_copy_replication, true, "Disable FREEZE PARTITION query for zero copy replication.", 0) \
     M(Bool, disable_detach_partition_for_zero_copy_replication, true, "Disable DETACH PARTITION query for zero copy replication.", 0) \
     M(Bool, disable_fetch_partition_for_zero_copy_replication, true, "Disable FETCH PARTITION query for zero copy replication.", 0) \
+    M(Bool, enable_block_number_column, false, "Enable persisting column _block_number for each row.", 0) ALIAS(allow_experimental_block_number_column) \
```
Comment (Algunenano) [inline src/Storages/MergeTree/MergeTreeSettings.h:189]: If these settings are not experimental anymore we should randomize them in CI. The last time that detected a number of issues IIRC and we had to remove it https://github.com/ClickHouse/ClickHouse/pull/56474

**Comment by clickgapai** [comment]
Hi @CurtizJ @alesapin — while reviewing this PR I found the following:

- **Coverage gap:** `SummingMergeTree` merge with `enable_block_offset_column=1` — `_block_offset` not-to-aggregate path untested — https://github.com/ClickHouse/ClickHouse/pull/103896

Could you review the test PR and add the `can be tested` label if it looks good?
Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**